cityofaustin/atd-moped
# Update component_arcgis_online_view database view to restore AGOL ETL

## Associated issues
Closes https://github.com/cityofaustin/atd-data-tech/issues/19915

This is a patch so I branched from `production` and will create another PR to move these changes into `main` after approvals.

[Here is more info](https://austininnovation.slack.com/archives/CNUEPKLB1/p1731632949125319?thread_ts=1731630011.967789&cid=CNUEPKLB1) about what this fixes.

## Testing
**URL to test:** 

Local only plz

**Steps to test:**
This is a tough one to test, but there is a way!

1. Edit your local stack and make the same [change shown in Frank's PR](https://github.com/cityofaustin/atd-moped/pull/1482/files). This will enable you to make a snapshot and replicate it from the new version of the production database (Postgres 16.5).
2. Now, check the `component_arcgis_online_view` in your database and see that every row in the `geometry` columns is either a `MultiLineString` or a `MultiPoint`.
3. Run the down migration and check again. You should see that there is a mixture of `MultiLineString`,`MultiPoint`, and `Point` type geometries.
4. Run the up migration again.
5. Now, you can run the AGOL ETL by going to `/atd-moped/moped-etl/arcgis`. You may need to set up an `env_file` as described in the readme. Make sure that you are pointing to your local graphql engine endpoint.
6. Run the AGOL ETL: `docker compose run arcgis -d`. This will update any project updated in the last week.
7. You should see the AGOL ETL succeed and log out some project IDs.
8. You can now compare the component geography of any of these IDs in the Moped production app to the [Moped components map](https://austin.maps.arcgis.com/apps/webappviewer/index.html?id=404d31d56b57491abe53ccfd718fcaee) when filtered to a project ID. You can filter in the left sidebar by scrolling towards the bottom.

**I have an old production snapshot from the Postgres 12.19 (pre-upgrade) DB that I used to compare the geo data formats. I can send that snapshot to anyone who wants to dig deeper.**

---
#### Ship list
